tests/console/zypper_ar.pm

```diff
@@ -22,10 +22,11 @@ sub run {
     if (my $urlprefix = get_var('MIRROR_PREFIX')) {
         my @repos_to_add = qw(OSS NON_OSS OSS_DEBUGINFO);
         my $repourl;
+        my $repos = script_output "zypper lr";
         foreach (@repos_to_add) {
             next unless get_var("REPO_$_");    # Skip repo if not defined
             $repourl = $urlprefix . "/" . get_var("REPO_$_");
-            zypper_call "ar -c $repourl $_" if zypper_call("lr | grep $_", allow_exit_codes => [1]);
+            zypper_call "ar -c $repourl $_" if ($repos !~ $_);    # Skip add repo if already added
```

Comment:
How about getting rid of the above `script_output` and instead executing the lookup for a potential already existing repo within the SUT context, e.g. something like

```perl
assert_script_run "zypper lr | grep $_ || zypper ar -c $repourl $_";
```